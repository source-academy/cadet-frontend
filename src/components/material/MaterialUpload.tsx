import { Divider } from '@blueprintjs/core';
import * as React from 'react';

import Dropzone from './Dropzone';
import { DirectoryData, MaterialData } from './materialShape';
import MaterialTable from './MaterialTable';

export interface IMaterialProps extends IDispatchProps, IStateProps {}

export interface IDispatchProps {
  handleCreateMaterialFolder: (title: string) => void;
  handleDeleteMaterial: (id: number) => void;
  handleDeleteMaterialFolder: (id: number) => void;
  handleFetchMaterialIndex: (id?: number) => void;
  handleUploadMaterial: (file: File, title: string, description: string) => void;
}

export interface IStateProps {
  materialDirectoryTree: DirectoryData[] | null;
  materialIndex: MaterialData[] | null;
}

class MaterialUpload extends React.Component<IMaterialProps, {}> {
  public render() {
    return (
      <div className="ContentDisplay row center-xs">
        <div className={`${'col-xs-10'} contentdisplay-content-parent`}>
          <Dropzone handleUploadMaterial={this.props.handleUploadMaterial} />
          <Divider />
          <MaterialTable
            handleCreateMaterialFolder={this.props.handleCreateMaterialFolder}
            handleDeleteMaterial={this.props.handleDeleteMaterial}
            handleDeleteMaterialFolder={this.props.handleDeleteMaterialFolder}
            handleFetchMaterialIndex={this.props.handleFetchMaterialIndex}
            materialDirectoryTree={this.props.materialDirectoryTree}
            materialIndex={this.props.materialIndex}
          />
        </div>
      </div>
    );
  }
}

export default MaterialUpload;
