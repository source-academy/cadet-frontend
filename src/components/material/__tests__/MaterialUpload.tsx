import { shallow } from 'enzyme';
import * as React from 'react';

import MaterialUpload, { IMaterialProps } from '../MaterialUpload';

test('Story Upload area renders correctly', () => {
  const props: IMaterialProps = {
    handleCreateMaterialFolder(p1: string) {},
    handleDeleteMaterial(p1: number) {},
    handleDeleteMaterialFolder(p1: number) {},
    handleFetchMaterialIndex(p1: number) {},
    handleUploadMaterial(p1: File, p2: string, p3: string) {},
    materialDirectoryTree: null,
    materialIndex: null
  };
  const app = <MaterialUpload {...props} />;
  const tree = shallow(app);
  expect(tree.debug()).toMatchSnapshot();
});
