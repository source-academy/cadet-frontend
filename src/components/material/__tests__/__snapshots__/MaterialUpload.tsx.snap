// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Upload area renders correctly 1`] = `
"<div className=\\"ContentDisplay row center-xs\\">
  <div className=\\"col-xs-10 contentdisplay-content-parent\\">
    <MaterialDropzone handleUploadMaterial={[Function: handleUploadMaterial]} />
    <Blueprint3.Divider />
    <MaterialTable handleCreateMaterialFolder={[Function: handleCreateMaterialFolder]} handleDeleteMaterial={[Function: handleDeleteMaterial]} handleDeleteMaterialFolder={[Function: handleDeleteMaterialFolder]} handleFetchMaterialIndex={[Function: handleFetchMaterialIndex]} materialDirectoryTree={{...}} materialIndex={{...}} />
  </div>
</div>"
`;
